vm/simple_server.go

```diff
@@ -172,16 +172,18 @@ func setupResponse(w http.ResponseWriter, req *http.Request, res *RObject) {
 		r.body = resBody.(*StringObject).Value
 	}
 
-	contentType, ok := res.instanceVariableGet("@content_type")
+	h, ok := res.instanceVariableGet("@headers")
 
-	if !ok {
-		r.contentType = "text/plain; charset=utf-8"
+	if headers, isHashObject := h.(*HashObject); ok && isHashObject {
+		for k, v := range headers.Pairs {
+			w.Header().Set(k, v.(*StringObject).Value)
```

Comment:
I think you need to check the value's type here and raise error if it's not String.